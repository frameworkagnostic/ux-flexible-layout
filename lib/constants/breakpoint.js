'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var BREAKPOINT_KEYS = exports.BREAKPOINT_KEYS = ['xs', 'sm', 'md', 'lg', 'xl'];

var BREAKPOINTS = exports.BREAKPOINTS = {
  xs: 0,
  sm: 600,
  md: 960,
  lg: 1280,
  xl: 1920
};
//# sourceMappingURL=breakpoint.js.map