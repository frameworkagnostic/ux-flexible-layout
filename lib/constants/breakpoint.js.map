{"version":3,"sources":["../../src/constants/breakpoint.js"],"names":["BREAKPOINT_KEYS","BREAKPOINTS","xs","sm","md","lg","xl"],"mappings":";;;;;AACO,IAAMA,4CAAkB,CAC7B,IAD6B,EAE7B,IAF6B,EAG7B,IAH6B,EAI7B,IAJ6B,EAK7B,IAL6B,CAAxB;;AAQA,IAAMC,oCAAc;AACzBC,MAAI,CADqB;AAEzBC,MAAI,GAFqB;AAGzBC,MAAI,GAHqB;AAIzBC,MAAI,IAJqB;AAKzBC,MAAI;AALqB,CAApB","file":"breakpoint.js","sourcesContent":["\nexport const BREAKPOINT_KEYS = [\n  'xs',\n  'sm',\n  'md',\n  'lg',\n  'xl'\n];\n\nexport const BREAKPOINTS = {\n  xs: 0,\n  sm: 600,\n  md: 960,\n  lg: 1280,\n  xl: 1920,\n};\n"]}