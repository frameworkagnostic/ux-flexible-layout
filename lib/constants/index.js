'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.GRID12_GUTTERS = exports.GRID12_SIZES = exports.STEP = exports.UNIT = exports.BREAKPOINTS = exports.BREAKPOINT_KEYS = undefined;

var _breakpoint = require('./breakpoint');

Object.defineProperty(exports, 'BREAKPOINT_KEYS', {
  enumerable: true,
  get: function get() {
    return _breakpoint.BREAKPOINT_KEYS;
  }
});
Object.defineProperty(exports, 'BREAKPOINTS', {
  enumerable: true,
  get: function get() {
    return _breakpoint.BREAKPOINTS;
  }
});

var _grid12Column = require('./grid12Column');

var grid12Column = _interopRequireWildcard(_grid12Column);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

var UNIT = exports.UNIT = 'px';
var STEP = exports.STEP = 5;

var GRID12_SIZES = exports.GRID12_SIZES = [].concat(_toConsumableArray(grid12Column.GRID_SIZES_DEFAULT), _toConsumableArray(grid12Column.GRID_SIZES));

var GRID12_GUTTERS = exports.GRID12_GUTTERS = grid12Column.GRID_GUTTERS;
//# sourceMappingURL=index.js.map