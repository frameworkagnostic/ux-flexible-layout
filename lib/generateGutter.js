'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var generateGutter = exports.generateGutter = function generateGutter(breakpoint, gutters) {
  var styles = {};

  gutters.forEach(function (spacing, index) {
    if (index === 0) {
      // Skip the default style.
      return;
    }

    styles['spacing-' + breakpoint + '-' + spacing] = {
      margin: -spacing / 2,
      width: 'calc(100% + ' + spacing + 'px)',
      '& > $item': {
        padding: spacing / 2
      }
    };
  });

  return styles;
};
//# sourceMappingURL=generateGutter.js.map