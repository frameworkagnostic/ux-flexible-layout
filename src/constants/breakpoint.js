
export const BREAKPOINT_KEYS = [
  'xs',
  'sm',
  'md',
  'lg',
  'xl'
];

export const BREAKPOINTS = {
  xs: 0,
  sm: 600,
  md: 960,
  lg: 1280,
  xl: 1920,
};
