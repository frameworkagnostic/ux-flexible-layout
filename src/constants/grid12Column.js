
export const GRID_GUTTERS = [
  0,
  8,
  16,
  24,
  32,
  40
];

export const GRID_SIZES_DEFAULT = [
  'auto',
  true,
];

export const GRID_SIZES = [
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
  11,
  12
];
