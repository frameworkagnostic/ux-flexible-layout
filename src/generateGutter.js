
export const generateGutter = (breakpoint, gutters) => {
  const styles = {};

  gutters.forEach((spacing, index) => {
    if (index === 0) {
      // Skip the default style.
      return;
    }

    styles[`spacing-${breakpoint}-${spacing}`] = {
      margin: -spacing / 2,
      width: `calc(100% + ${spacing}px)`,
      '& > $item': {
        padding: spacing / 2,
      },
    };
  });

  return styles;
}
