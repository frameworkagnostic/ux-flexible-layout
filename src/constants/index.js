
import * as grid12Column from './grid12Column';

export {
  BREAKPOINT_KEYS,
  BREAKPOINTS,
} from './breakpoint';

export const UNIT = 'px';
export const STEP = 5;

export const GRID12_SIZES = [
  ...grid12Column.GRID_SIZES_DEFAULT,
  ...grid12Column.GRID_SIZES,
];

export const GRID12_GUTTERS = grid12Column.GRID_GUTTERS;
