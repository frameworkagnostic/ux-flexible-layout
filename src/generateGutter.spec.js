
import { generateGutter } from './generateGutter';
import { GRID12_GUTTERS } from './constants';

describe('generate gutter', () => {
  test('isFunction', () => expect(generateGutter).toEqual(expect.any(Function)));

  describe('gutter 12 columns', () => {
    test('snapshot', () => expect(GRID12_GUTTERS).toMatchSnapshot());

    describe('create styles', () => {
      const styles = generateGutter('xs', GRID12_GUTTERS);
      test('isObject', () => expect(styles).toEqual(expect.any(Object)));
      test('snapshot', () => expect(styles).toMatchSnapshot());
    });
  });
});
