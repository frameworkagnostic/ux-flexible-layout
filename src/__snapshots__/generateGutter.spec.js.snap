// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate gutter gutter 12 columns create styles snapshot 1`] = `
Object {
  "spacing-xs-16": Object {
    "& > $item": Object {
      "padding": 8,
    },
    "margin": -8,
    "width": "calc(100% + 16px)",
  },
  "spacing-xs-24": Object {
    "& > $item": Object {
      "padding": 12,
    },
    "margin": -12,
    "width": "calc(100% + 24px)",
  },
  "spacing-xs-32": Object {
    "& > $item": Object {
      "padding": 16,
    },
    "margin": -16,
    "width": "calc(100% + 32px)",
  },
  "spacing-xs-40": Object {
    "& > $item": Object {
      "padding": 20,
    },
    "margin": -20,
    "width": "calc(100% + 40px)",
  },
  "spacing-xs-8": Object {
    "& > $item": Object {
      "padding": 4,
    },
    "margin": -4,
    "width": "calc(100% + 8px)",
  },
}
`;

exports[`generate gutter gutter 12 columns snapshot 1`] = `
Array [
  0,
  8,
  16,
  24,
  32,
  40,
]
`;
